src/adapter/src/coord/sequencer/inner/create_materialized_view.rs

```diff
@@ -285,6 +281,40 @@ impl Coordinator {
         )
         .collect::<Vec<_>>();
 
+        // Timestamp selection
+        let as_of = {
+            // Normally, `as_of` should be the least_valid_read.
+            let id_bundle = dataflow_import_id_bundle(global_lir_plan.df_desc(), cluster_id);
+            let mut as_of = self.least_valid_read(&id_bundle);
+            // But for MVs with non-trivial REFRESH schedules, it's important to set the
+            // `as_of` to the first refresh. This is because we'd like queries on the MV to
+            // block until the first refresh (rather than to show an empty MV).
+            if let Some(refresh_schedule) = &refresh_schedule {
+                if let Some(ts) = as_of.as_option() {
+                    let Some(rounded_up_ts) = refresh_schedule.round_up_timestamp(*ts) else {
+                        return Err(AdapterError::MaterializedViewWouldNeverRefresh(name.item));
+                    };
+                    as_of = Antichain::from_elem(rounded_up_ts);
+                } else {
+                    // The `as_of` should never be empty, because then the MV would be unreadable.
+                    soft_panic_or_log!("creating a materialized view with an empty `as_of`");
+                }
+            }
+            as_of
+        };
+
+        // If we have a refresh schedule that has a last refresh, then set the `until` to the last refresh.
+        // (If the `try_step_forward` fails, then no need to set an `until`, because it's not possible to get any data
+        // beyond that last refresh time, because there are no times beyond that time.)
+        let until = refresh_schedule
+            .map(|refresh_schedule| {
+                refresh_schedule
+                    .last_refresh()
+                    .map(|last_refresh| last_refresh.try_step_forward())
+            })
+            .flatten()
+            .flatten();
```

Comment:
Same here:

```suggestion
        let until = refresh_schedule
            .and_then(|s| s.last_refresh())
            .and_then(|r| r.try_step_forward());
```

Reply:
Fixed in 1f4b5af3b8792d5bf05230d20dcd04a70dce827b